### PR TITLE
Swift HTTP Types

### PR DESCRIPTION
**Pull request by adam-fowler**

### CODE REVIEW

File: Sources/Hummingbird/Middleware/TracingMiddleware.swift [hunk 1]
```diff
@@ -53,10 +54,11 @@ public struct HBTracingMiddleware<Context: HBBaseRequestContext>: HBMiddleware {
             span.updateAttributes { attributes in
                 attributes["http.method"] = request.method.rawValue
                 attributes["http.target"] = request.uri.path
-                attributes["http.flavor"] = "\(request.version.major).\(request.version.minor)"
-                attributes["http.scheme"] = request.uri.scheme?.rawValue
-                attributes["http.user_agent"] = request.headers.first(name: "user-agent")
-                attributes["http.request_content_length"] = request.headers["content-length"].first.map { Int($0) } ?? nil
+                // TODO: Get HTTP version and scheme
```
Comment (Joannis) [inline Sources/Hummingbird/Middleware/TracingMiddleware.swift:57]: Commenting to mark this TODO

File: Sources/Hummingbird/Exports.swift [hunk 2]
```diff
@@ -25,6 +25,5 @@
 @_exported import struct NIOCore.ByteBuffer
 @_exported import struct NIOCore.ByteBufferAllocator
 
-@_exported import struct NIOHTTP1.HTTPHeaders
-@_exported import enum NIOHTTP1.HTTPMethod
-@_exported import enum NIOHTTP1.HTTPResponseStatus
+@_exported import struct HTTPTypes.HTTPRequest
+@_exported import struct HTTPTypes.HTTPResponse
```
Comment (Joannis) [inline Sources/Hummingbird/Exports.swift:29]: Shouldn't we also export `HTTPFields`?

File: Sources/Hummingbird/Middleware/TracingMiddleware.swift (hunk 1 above)
Comment (adam-fowler) [reply Sources/Hummingbird/Middleware/TracingMiddleware.swift:57]: We never setup the scheme before, it always defaulted to `nil`. The swift http types don't include a version so not sure how we are going to get that

File: Package.swift [hunk 3]
```diff
@@ -7,17 +7,19 @@ let package = Package(
...
         .library(name: "HummingbirdHTTP2", targets: ["HummingbirdHTTP2"]),
-        .library(name: "HummingbirdFoundation", targets: ["HummingbirdFoundation"]),
-        .library(name: "HummingbirdJobs", targets: ["HummingbirdJobs"]),
-        .library(name: "HummingbirdXCT", targets: ["HummingbirdXCT"]),
-        .executable(name: "PerformanceTest", targets: ["PerformanceTest"]),
+        .library(name: "HummingbirdTLS", targets: ["HummingbirdTLS"]),
+        // .library(name: "HummingbirdFoundation", targets: ["HummingbirdFoundation"]),
+        // .library(name: "HummingbirdJobs", targets: ["HummingbirdJobs"]),
+        // .library(name: "HummingbirdXCT", targets: ["HummingbirdXCT"]),
+        // .executable(name: "PerformanceTest", targets: ["PerformanceTest"]),
```
Comment (Joannis) [inline Package.swift:17]: Any particular reason for lal these disabled products?

File: Package.swift [hunk 4]
```diff
@@ -7,17 +7,19 @@ let package = Package(
     name: "hummingbird",
     platforms: [.macOS(.v14), .iOS(.v17), .tvOS(.v17)],
     products: [
-        .library(name: "Hummingbird", targets: ["Hummingbird"]),
+        // .library(name: "Hummingbird", targets: ["Hummingbird"]),
```
Comment (Joannis) [inline Package.swift:10]: Disabled product

**Review by Joannis — APPROVED** [review]

**Comment by Joannis** [comment]
What's changed looks good, but we have disabled some products. Can we get those back up & running?

**Review by Joannis — CHANGES_REQUESTED** [review]
LGTM, but disabled products

**Comment by adam-fowler** [comment]
> LGTM, but disabled products

I must have disabled them as I was getting stuff compiling.

**Pull request merged**